src/main/java/net/minecraftforge/event/AnvilUpdateEvent.java

```diff
@@ -19,52 +19,146 @@
-    @Nonnull
-    private ItemStack output;      // Set this to set the output stack
-    private int cost;              // The base cost, set this to change it if output != null
-    private int materialCost; // The number of items from the right slot to be consumed during the repair. Leave as 0 to consume the entire stack.
-
-    public AnvilUpdateEvent(@Nonnull ItemStack left, @Nonnull ItemStack right, String name, int cost)
+public class AnvilUpdateEvent extends Event {
+
+    private final ItemStack left;
+    private final ItemStack right;
+    private final String name;
+    private ItemStack output;
+    private int cost;
+    private int materialCost;
+    @Nullable // TODO: Remove 1.17 - Nullable only in the instance that a mod uses the deprecated ctor.
+    private final PlayerEntity player;
+
+    @Deprecated //TODO: Remove 1.17 - Use Player-contextual constructor below.
+    public AnvilUpdateEvent(ItemStack left, ItemStack right, String name, int cost)
+    {
+        this(left, right, name, cost, null);
+    }
+
+    public AnvilUpdateEvent(ItemStack left, ItemStack right, String name, int cost, PlayerEntity player)
     {
         this.left = left;
         this.right = right;
         this.output = ItemStack.EMPTY;
         this.name = name;
+        this.player = player;
         this.setCost(cost);
         this.setMaterialCost(0);
     }
 
-    @Nonnull
-    public ItemStack getLeft() { return left; }
-    @Nonnull
-    public ItemStack getRight() { return right; }
-    public String getName() { return name; }
-    @Nonnull
-    public ItemStack getOutput() { return output; }
-    public void setOutput(@Nonnull ItemStack output) { this.output = output; }
-    public int getCost() { return cost; }
-    public void setCost(int cost) { this.cost = cost; }
-    public int getMaterialCost() { return materialCost; }
-    public void setMaterialCost(int materialCost) { this.materialCost = materialCost; }
+    /**
+     * @return ihe Item in the left input (leftmost) anvil slot.
+     */
+    public ItemStack getLeft()
+    {
+        return left;
+    }
+
+    /**
+     * @return The item in the right input (center) anvil slot.
+     */
+    public ItemStack getRight()
+    {
+        return right;
+    }
+
+    /**
+     * This is the name as sent by the client.  It may be null if none has been sent. <br>
+     * If empty, it indicates the user wishes to clear the custom name from the item.
+     * @return The name that the output item will be set to, if applicable.
+     */
+    @Nullable
+    public String getName()
+    {
+        return name;
+    }
+
+    /**
+     * This is the output as a result of this event, not of the two items. <br>
+     * If you are the first receiver of this event, it is guaranteed to be empty. <br>
+     * It will only be non-empty if changed by an event handler.
+     * @return The item in the output (rightmost) anvil slot.
```

Comment:
Given the above description, the label for the return value is wrong (it's the item that should be put in that slot, not the one currently in it).